src/lib/filter/adjudicators/adjudicators.ts

```diff
@@ -73,10 +73,13 @@ export const carriedName = pipe(
 export const carriesName = pipe(carriedName, equals(""), not);
 export const missingName = complement(carriesName);
 
-export const carriedNamespace = pipe(
-  (kubernetesObject: KubernetesObject): string | undefined => kubernetesObject?.metadata?.namespace,
-  defaultTo(""),
-);
+export const carriedNamespace = pipe((kubernetesObject: KubernetesObject): string | undefined => {
+  if (kubernetesObject?.kind === "Namespace") {
```

Comment:
This looks like a hold-over from the first attempt, which... shouldn't be needed any longer, right?

Since we decided that the `carried[...]()` methods were just data accessors that _shouldn't_ behave in a conditional way & you've already moved the "is it a Namespace" checks over into `uncarryableNamespace()` and `carriesIgnoredNamespace()`..?

Reply:
If we do not include this, then we will not have an accurate filter message

```plaintext
Ignoring Watch Callback: Object carries namespace '' but ignored namespaces include '["pepr-demo-2"]'.
```

```ts
export function adjudicateUncarryableNamespace(
  capabilityNamespaces: string[],
  obj: KubernetesObject,
): AdjudicationResult {
  return uncarryableNamespace(capabilityNamespaces, obj)
    ? `Object carries namespace '${carriedNamespace(obj)}' but namespaces allowed by Capability are '${JSON.stringify(capabilityNamespaces)}'.`
    : null;
}
```

Reply:
unless of course we have a 1-off check inside of `adjudicateUncarryableNamespace`, is that what you prefer here?

A check inside of that function?

Reply:
Okay, i accounted for it in the `adjudicate**` funcs and changed it back